### PR TITLE
SONARSEC-199: Handle event registration

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs [hunk 1]
```diff
@@ -577,6 +577,11 @@ private IEnumerable<Instruction> ProcessAssignmentExpression(AssignmentExpressio
                 return instructions;
             }
 
+            if (leftPartSymbol is IEventSymbol)
+            {
+                return instructions;
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs]: When the assignment is on an event we don't need to do anything.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/UcfgInstructionTest.cs [hunk 2]
```diff
@@ -862,5 +862,37 @@ public override bool Equals(object obj)
...
+        {
+            var e = new SocketAsyncEventArgs { UserToken = null };
+                // %0 := new System.Net.Sockets.SocketAsyncEventArgs
+                // %1 := System.Net.Sockets.SocketAsyncEventArgs.SocketAsyncEventArgs() [ %0 ]
+                // %2 := System.Net.Sockets.SocketAsyncEventArgs.UserToken.set [ %0 const ]
+                // e := __id [ %0 ]
+
+            e.Completed += _onComplete;
+                // %3 := __id [ this._onComplete ]
+                // %4 := System.Net.Sockets.SocketAsyncEventArgs.Completed.add [ System.Net.Sockets.SocketAsyncEventArgs const %3 ]
```
Comment (duncanp-sonar) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/UcfgInstructionTest.cs]: It seems a bit odd that there isn't a reference to _e_ in %4. Still, I guess there's no way adding the event handler could taint _e_ (i.e. if _this_ was tainted).

Comment (Evangelink) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/UcfgInstructionTest.cs]: I actually didn't noticed that `e` was missing. It's more corner case so I will ignore it for now but create a ticket about it.
